File: src/api/system/publicFile.ts
import { post } from '@/utils/request'

export function getList<T>(type?: string) {
  return post<T>({
    url: '/publicFile/getList',
    data: type ? { type } : {},
  })
}

export function deletes<T>(ids: number[]) {
  return post<T>({
    url: '/publicFile/deletes',
    data: { ids },
  })
}

export function updateType<T>(id: number, type: string) {
  return post<T>({
    url: '/publicFile/updateType',
    data: { id, type },
  })
}

File: service/router/system/publicFile.go
package system

import (
	"panel-next/api/api_v1"
	"panel-next/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitPublicFileRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiSystem.PublicFileApi

	admin := router.Group("", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		admin.POST("/publicFile/upload", api.Upload)
		admin.POST("/publicFile/deletes", api.Deletes)
		admin.POST("/publicFile/updateType", api.UpdateType)
	}

	// 所有登录用户可读
	public := router.Group("", middleware.LoginInterceptor)
	{
		public.GET("/publicFile/getList", api.GetList)
	}
}
